### PR TITLE
test: Add e2e testing for deprecated AMIs

### PR DESCRIPTION
**Pull request by engedaam**
Fixes #N/A 

**Description**

**How was this change tested?**

**Does this change impact docs?**
- [ ] Yes, PR includes docs updates 
- [ ] Yes, issue opened: # 
- [x] No

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

**Review by engedaam — COMMENTED** [review]
/karpenter snapshot

**Review by engedaam — COMMENTED** [review]
/karpenter snapshot

**Review by engedaam — COMMENTED** [review]
/karpenter snapshot

File: test/pkg/environment/aws/expectations.go [hunk 1]
```diff
@@ -362,6 +362,55 @@ func (env *Environment) GetAMIBySSMPath(ssmPath string) string {
...
+		IncludeDeprecated: lo.ToPtr(true),
+	})
+	Expect(err).To(BeNil())
+	if len(out.Images) == 1 {
+		return lo.FromPtr(out.Images[0].ImageId)
+	}
+
+	input := &ec2.CopyImageInput{
+		SourceImageId: lo.ToPtr(amiID),
+		Name:          lo.ToPtr(fmt.Sprintf("depricated-%s-%s-%s", amiID, amifamily, env.ClusterVersion)),
```
Comment (rschalo) [inline test/pkg/environment/aws/expectations.go:386]:
```suggestion
		Name:          lo.ToPtr(fmt.Sprintf("deprecated-%s-%s-%s", amiID, amifamily, env.ClusterVersion)),
```

File: test/pkg/environment/aws/environment.go [hunk 2]
```diff
@@ -72,6 +75,7 @@ type Environment struct {
 
 	ClusterName       string
 	ClusterEndpoint   string
+	ClusterVersion    string
```
Comment (rschalo) [inline test/pkg/environment/aws/environment.go:78]: Does using `env.K8sVersion()` not work for these tests?

Comment (engedaam) [reply test/pkg/environment/aws/environment.go]: Yeah, that would work here. Didn't see that

File: test/pkg/environment/aws/expectations.go [hunk 3]
```diff
@@ -34,7 +34,7 @@ import (
 	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
 	"k8s.io/apimachinery/pkg/types"
 
-	coretest "sigs.k8s.io/karpenter/pkg/test"
+	"sigs.k8s.io/karpenter/pkg/test"
```
Comment (rschalo) [inline test/pkg/environment/aws/expectations.go:37]: Can you make `coretest` to reduce diff here?

File: test/suites/drift/suite_test.go [hunk 4]
```diff
@@ -383,26 +384,32 @@ var _ = Describe("Drift", func() {
...
-	It("should not disrupt nodes that have drifted without the featureGate enabled", func() {
-		env.ExpectSettingsOverridden(corev1.EnvVar{Name: "FEATURE_GATES", Value: "Drift=false"})
 
-		oldCustomAMI := env.GetAMIBySSMPath(fmt.Sprintf("/aws/service/eks/optimized-ami/%s/amazon-linux-2023/x86_64/standard/recommended/image_id", env.K8sVersionWithOffset(1)))
+		// validate the AMI id matches the non-deprecated AMI
+		pod = env.EventuallyExpectHealthyPodCount(selector, numPods)[0]
+		env.ExpectInstance(pod.Spec.NodeName).To(HaveField("ImageId", HaveValue(Equal(amdAMI))))
+
+	})
+	It("should return drifted if the AMI no longer matches the existing NodeClaims instance type", func() {
```
Comment (rschalo) [inline test/suites/drift/suite_test.go:393]: Does this test become longer than necessary if we assert the AMI has changed on the new nodeclaim?

Comment (engedaam) [reply test/suites/drift/suite_test.go:393]: We need to assert that we have a new nodeclaim, no?

Comment (rschalo) [reply test/suites/drift/suite_test.go:420]: Yea, I think the other test covers the new AMI case.

**Review by rschalo — APPROVED** [review]
LGTM

**Pull request merged**